### PR TITLE
Rename project to Stytch from StytchDemo.

### PR DESCRIPTION
**Pull request by nidal-stytch**
## Changes:

1. Rename the project file!!
2. Also a general cleanup of the project file and delete a bunch of stale files.
3. Delete the server group
4. Move sample apps into a group

## Checklist:
- [x] I have verified that this change works in the relevant demo app, or N/A
- [x] I have added or updated any tests relevant to this change, or N/A
- [x] I have updated any relevant README files for this change, or N/A

### CODE REVIEW

**Comment by jhaven-stytch** [comment]
I'm totally down with removing the stale files, but I'm not sure I'm onboard with the rename of StytchDemo/StytchUIDemo. What's the impact/impetus for doing so? I think calling them Stytch/StytchUI is confusing because those projects are specifically _not_ those packages, they _are_ demos of those packages/sources. Lets chat.

File: Makefile [hunk 1]
```diff
@@ -6,8 +6,8 @@ IS_CI=$(shell [ ! -z "$$CI" ] && echo "1")
 ARCH=arch -$(shell [ $(IS_CI) ] && echo "x86_64" || echo "arm64")
 PIPEFAIL=set -o pipefail
 XCPRETTY=bundle exec xcpretty
-TEST=$(PIPEFAIL) && $(ARCH) xcodebuild test -disableAutomaticPackageResolution -skipPackageUpdates -project StytchDemo/StytchDemo.xcodeproj -scheme StytchCoreTests -sdk
```
Comment (jhaven-stytch) [inline Makefile:9]: Is there a reason/benefit to removing the `-disableAutomaticPackageResolution` flag?

File: Scripts/demo [hunk 2]
```diff
@@ -41,7 +41,7 @@ class Runner
   def cleanup
     puts 'Cleaning up server'
     system('pkill ngrok')
-    system('pkill \'StytchDemo \(Server\)\'')
```
Comment (jhaven-stytch) [inline Scripts/demo:44]: Since we killed `make demo`, I think we can delete this file

File: Stytch/DemoApps/StytchDemo/Shared/Info.plist [hunk 3]
```diff
@@ -10,7 +10,7 @@
 			<key>CFBundleTypeRole</key>
 			<string>Editor</string>
 			<key>CFBundleURLName</key>
-			<string>com.stytch.StytchDemo</string>
+			<string>com.stytch.Stytch</string>
```
Comment (jhaven-stytch) [inline Stytch/DemoApps/StytchDemo/Shared/Info.plist:13]: I feel like maybe the bundle ID should stay as `StytchDemo`?

**Review by jhaven-stytch — APPROVED** [review]
Couple questions, but otherwise, seems cleaner 👍

File: Stytch/DemoApps/StytchDemo/Shared/Info.plist (hunk 3 above)
Comment (nidal-stytch) [reply Stytch/DemoApps/StytchDemo/Shared/Info.plist]: fixed

File: Scripts/demo (hunk 2 above)
Comment (nidal-stytch) [reply Scripts/demo:44]: fixed

File: Makefile (hunk 1 above)
Comment (nidal-stytch) [reply Makefile:9]: I was getting errors on this for some reason so I deleted it

Comment (nidal-stytch) [reply Makefile:9]: ok added it back and fixed the ignore file.

**Pull request merged**